internal/flink/command_shell.go

```diff
@@ -70,14 +72,10 @@ func (c *command) authenticated(authenticated func(*cobra.Command, []string) err
 }
 
 func (c *command) startFlinkSqlClient(prerunner pcmd.PreRunner, cmd *cobra.Command) error {
-	if useFakeGateway {
-		return client.StartApp(
-			mock.NewFakeFlinkGatewayClient(),
-			func() error { return nil },
-			types.ApplicationOptions{
-				Context:   c.Context,
-				UserAgent: c.Version.UserAgent,
-			}, func() {})
+	// if config-keys were passed, we should enter local mode
+	configKeys, _ := cmd.Flags().GetStringSlice("config-key")
+	if len(configKeys) > 0 {
+		return c.startWithLocalMode(cmd)
 	}
```

Comment:
Instead of ignoring this error, consider wrapping this block with the feature flag?